adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/BrokerProxy.cs

```diff
@@ -364,17 +364,13 @@ public Intent GetIntentForBrokerActivity(AuthenticationRequest request, Activity
                     intent.PutExtra(BrokerConstants.BrokerRequest, BrokerConstants.BrokerRequest);
                 }
             }
-            catch (OperationCanceledException e)
-            {
-                RequestContext.Logger.Error(e);
-                RequestContext.Logger.ErrorPii(e);
-            }
             catch (Exception e)
             {
                 // Authenticator gets problem from webrequest or file read/write
                 var ex = new AdalException("Authenticator cancels the request", e);
                 RequestContext.Logger.Error(ex);
                 RequestContext.Logger.ErrorPii(ex);
+                throw ex;
```

Comment:
>throw ex; [](start = 16, length = 9)

was this a previous miss or is this an intentional behavior change? Is this handled anywhere in our code or do we pass this exception to the user?

Reply:
@SomkaPe and I looked this over. It is an intentional behavior change. The exception is passed to the user.

Reply:
shouldn't this be in the above now removed catch intead. and we should leave the generic exception alone?

---
In reply to: [205926366](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1140#discussion_r205926366) [](ancestors = 205926366)

Reply:
The `OperationCancelledException` was just suppressed and written to the logs. I think we should catch the exceptions and throw an AdalException.

Reply:
yes, but you are now catching all exceptions and not just the specific one regarding operationcanceledexception... imo this code should be moved up to what was there before. and a general throw should be put in place for the general exception. (or it should not be handled at all)

---
In reply to: [205926683](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1140#discussion_r205926683) [](ancestors = 205926683)

Reply:
addressed all PR comments and tested